pkg/utils/secureopen/secureopen.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package secureopen provides a way to securely open a file in a container and
+// checking that the path didn't move outside of the container rootfs.
+package secureopen
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/utils/host"
+)
+
+// OpenInContainer opens the given path in the given container in read-only
+// mode.
```

Comment:
```suggestion
// OpenInContainer opens the given path in the given container referenced by containerPid in read-only
// mode.
```

btw, this is more generic than containers (this works for any process), but I think leaving it for containers is okay here.